python-sdk/tutorials/automl-with-azureml/forecasting-pipelines/scripts/infer.py

```diff
@@ -0,0 +1,210 @@
+    )
+
+    parser.add_argument(
+        "--time_column_name",
+        type=str,
+        dest="time_column_name",
+        help="The time column name.",
+    )
+
+    parser.add_argument(
+        "--max_horizon",
+        type=str,
+        dest="max_horizon",
+        help="The max horizon.",
+    )
+
+    parser.add_argument(
+        "--ouput_dataset_name",
+        type=str,
+        dest="ouput_dataset_name",
+        default="results",
+        help="Dataset name of the final output",
+    )
+    parser.add_argument(
+        "--test_dataset_name",
+        type=str,
+        dest="test_dataset_name",
+        default="results",
+        help="Dataset name of the final output",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def get_data(
+    run,
+    fitted_model,
+    target_column_name,
+    time_column_name,
+    max_horizon,
+    freq,
+    test_dataset_name,
+):
+
+    # get input dataset by name
+    test_dataset = Dataset.get_by_name(run.experiment.workspace, test_dataset_name)
+    test_df = test_dataset.to_pandas_dataframe()
+    if target_column_name in test_df:
+        y_test = test_df.pop(target_column_name)
+    else:
+        y_test = np.full(test_df.shape[0], np.nan)
+
+    return test_df, y_test
+
+
+def get_freq_from_model(fitted_model):
+    try:
+        freq = fitted_model.data_frequency
+    except:
+        freq = fitted_model._pre_transform.freq_offset
+    return freq
+
+
+def get_model_filename(run, model_name, model_path):
+    model = Model(run.experiment.workspace, model_name)
+    if "model_file_name" in model.tags:
+        return model.tags["model_file_name"]
+    is_pkl = True
+    if model.tags.get("algorithm") == "TCNForecaster" or os.path.exists(
+        os.path.join(model_path, "model.pt")
+    ):
+        is_pkl = False
+    return "model.pkl" if is_pkl else "model.pt"
+
+
+if __name__ == "__main__":
+    run = Run.get_context()
+
+    args = get_args()
+    model_name = args.model_name
+    ouput_dataset_name = args.ouput_dataset_name
+    test_dataset_name = args.test_dataset_name
+    target_column_name = args.target_column_name
+    time_column_name = args.time_column_name
+    max_horizon = args.max_horizon
+    print("args passed are: ")
+
+    print(model_name)
+    print(test_dataset_name)
+    print(ouput_dataset_name)
+    print(target_column_name)
+
+    model_path = Model.get_model_path(model_name)
+    model_file_name = get_model_filename(run, model_name, model_path)
+    print(model_file_name)
+    fitted_model = get_model(model_path, model_file_name)
+
+    print("model related params are:")
+    print(max_horizon)
+    print(time_column_name)
+
+    freq = get_freq_from_model(fitted_model)
+
+    X_test_df, y_test = get_data(
+        run,
+        fitted_model,
+        target_column_name,
+        time_column_name,
+        max_horizon,
+        freq,
+        test_dataset_name,
+    )
+
+    df = infer_forecasting_dataset_tcn(
```

Comment:
May e we want to use Pipeline output to save the data frame obtained from prediction?

Reply:
added